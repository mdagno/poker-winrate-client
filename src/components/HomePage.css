.homePage{
  color: #222;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 80%;
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
  border-radius: 3px;
  border-top: 10px outset white;
  border-left: 10px outset white;
  background-color: #ccc;
  text-align: left;
}

.homePage h2 {
  text-align: left;
  margin: 0;
  padding: 0.5em;
  padding-bottom: 0.2em;
  border-bottom: 2px dashed white;
}

.homePage h3{
  margin: 0;
  padding: 1em;
  border-bottom: 1px solid white;
}

@media only screen and (min-device-width : 768px) and (max-device-width : 1024px){
  .homePage{
    text-align: center;
  }
  
}

/* Desktops and laptops ----------- */
@media only screen  and (min-width : 1224px) {
  .homePage{
    text-align: center;
    width: 40%;
    height: 50%;
    border-top: 30px outset white;
    border-left: 30px outset white;
  }
  
  .homePage h2{
    padding: 1em;
    margin-left: 1em;
  }
}