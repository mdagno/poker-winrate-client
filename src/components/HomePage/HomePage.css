.homePage{
  color: #222;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 80%;
  margin-top: 50px;
  margin-left: auto;
  margin-right: auto;
  border-radius: 3px;
  border-top: 10px outset white;
  border-left: 10px outset white;
  background-color: #ccc;
  text-align: left;
}

.homePage h2 {
  text-align: left;
  margin: 0;
  padding: 0.5em;
  padding-bottom: 0.2em;
  border-bottom: 2px dashed white;
  font-style: italic;
}

.homePage h3{
  margin: 0;
  padding: 1em;
  border-bottom: 1px solid white;
  font-size: 1em;
}

@media only screen and (min-device-width : 768px){
  .homePage{
    text-align: center;
    border-top: 30px outset white;
    border-left: 30px outset white;
    margin-top: 150px;
    width: 70%;
  }
  
  .homePage h2{
    padding: 1em;
    margin-left: 1em;
  }
}

@media only screen and (min-device-width : 1024px){
  .homePage{
    text-align: center;
    width: 40%;
    height: 50%;
    margin-top: 115px;
  }
  
  .homePage h2{
    padding: 1em;
    margin-left: 1em;
  }
}