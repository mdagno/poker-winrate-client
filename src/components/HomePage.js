import React from 'react';
import ApiContext from '../ApiContext';
import './HomePage.css';

export default class HomePage extends React.Component{
  static contextType = ApiContext;

  calculateTotalEarnings = () => {
    let totalEarnings = 0;
    this.context.sessionsList.forEach(session => {
      let sessionEarnings = session.cashed_out - session.buy_in;
      totalEarnings = totalEarnings + sessionEarnings;
    })

    return totalEarnings;
  }

  computeTotalSessions = () => {
    return this.context.sessionsList.length;
  }

  calculateWinRate = () => {
    let sessionWinRates = this.context.sessionsList.map(session => {
      return (session.cashed_out - session.buy_in) / session.session_length / session.big_blind
    })
    console.log(sessionWinRates)
    let sessionWinRatesSum = sessionWinRates.reduce((a, b) => (a + b), 0)
    return (sessionWinRatesSum / sessionWinRates.length).toFixed(2)
  }
  
  
  render(){
    return(
      <div className='homePage'>
        <h3>Net Earnings: ${this.calculateTotalEarnings()}</h3>
        <h3>Most Played Stake: $5/$10</h3>
        <h3>Total Sessions: {this.computeTotalSessions()}</h3>
        <h3>{this.calculateWinRate()} BB/hr</h3>
      </div>
  )
  }
}