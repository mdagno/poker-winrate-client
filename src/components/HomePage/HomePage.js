import React from 'react';
import ApiContext from '../../ApiContext';
import './HomePage.css';

export default class HomePage extends React.Component{
  static contextType = ApiContext;

  calculateCashGameProfits = () => {
    let totalEarnings = 0;
    let cashWinnings = this.context.sessionsList.filter(session => session.game_type_two === 'Cash')
    cashWinnings.forEach(session => {
      let sessionEarnings = session.cashed_out - session.buy_in;
      totalEarnings = totalEarnings + sessionEarnings;
    })

    return totalEarnings;
  }

  calculateTournamentProfits = () => {
    let totalEarnings = 0;
    let tournamentWinnings = this.context.sessionsList.filter(session => session.game_type_two === 'Tournament')
    tournamentWinnings.forEach(session => {
      let sessionEarnings = session.cashed_out - session.buy_in;
      totalEarnings = totalEarnings + sessionEarnings;
    })

    return totalEarnings;
  }

  calculateWinRate = () => {
    let cashGames = this.context.sessionsList.filter(session => session.game_type_two === 'Cash')
    let sessionWinRates = cashGames.map(session => {
      return (session.cashed_out - session.buy_in) / session.session_length / session.big_blind
    })
    let sessionWinRatesSum = sessionWinRates.reduce((a, b) => (a + b), 0)
    return (sessionWinRatesSum / sessionWinRates.length).toFixed(2)
  }

  calculateMode = (arr) => {
    let mf = 1;
    let count = 0;
    let mode;

    for (let i = 0; i < arr.length; i++) {
      for (var j = i; j < arr.length; j++){
        if (arr[i] == arr[j]){
          count++;
          if(mf < count){
            mf = count;
            mode = arr[i]
          }
        }
      }
      count = 0;
    }
    return mode;
  }

  // constructBlindValueKey = (session) => {
  //   return `${session.small_blind}  ${session.big_blind}`;
  // }

  // deconstructBlindValueKey = (string) => {
  //   return string.split(' ');
  // }

  // calculateMostPlayedStake = () => {
  //   let trackerObj = {};
  //   let maxKey;
  //   let maxCount;
    
  //   this.context.sessionsList.forEach(session => {
  //      const key = this.constructBlindValueKey(session)
  //      let count = 1;
  //      if(!trackerObj[key]){
  //       trackerObj[key] = count;
  //       maxKey = key;
  //       maxCount = count;
  //      }
  //      else{
  //        trackerObj[key]++;
  //        count = trackerObj[key];
  //        if(count > maxKey[key]){
  //         maxKey = key;
  //         maxCount = count;
  //        }
  //      }
  //   })



  //   return maxKey;
  // }


 

  // calculateMostPlayedStake = () => {
  //   let sessionsList = this.context.sessionsList;
  //   let sbValues = sessionsList.map(session => {
  //     return session.small_blind
  //   })
  //   let bbValues = sessionsList.map(session => {
  //     return session.big_blind
  //   })
    
  //   return [this.calculateMode(sbValues), this.calculateMode(bbValues)]
  // }
  
  render(){
    return(
      <div className='homePage'>
        <h2>Your Summary</h2>
        <h3>${this.calculateCashGameProfits() + this.calculateTournamentProfits()} Total Profit</h3>
        <h3>${this.calculateCashGameProfits()} Cash Game Winnings</h3>
        <h3>${this.calculateTournamentProfits()} Tournament Winnings</h3>
        {/* <h3>Most Played Stake: ${this.calculateMostPlayedStake()[0]} / ${this.calculateMostPlayedStake()[1]}</h3> */}
        <h3>{this.context.sessionsList.length} Sessions Played</h3>
        <h3>{this.calculateWinRate()} BB/hr</h3>
      </div>
  )
  }
}