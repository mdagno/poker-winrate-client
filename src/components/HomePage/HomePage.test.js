import React from 'react';
import HomePage from './HomePage';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

describe('HomePage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HomePage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('HomePage renders correctly', () => {
    const tree = renderer
      .create(<HomePage />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
