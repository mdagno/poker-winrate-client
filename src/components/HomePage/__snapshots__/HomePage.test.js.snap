// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component HomePage renders correctly 1`] = `
<div
  className="homePage"
>
  <h2>
    Your Summary
  </h2>
  <h3>
    $
    0
     Total Profit
  </h3>
  <h3>
    $
    0
     Cash Game Winnings
  </h3>
  <h3>
    $
    0
     Tournament Winnings
  </h3>
  <h3>
    0
     Sessions Played
  </h3>
  <h3>
    NaN
     BB/hr
  </h3>
</div>
`;
